meetings/2022/notes-2022-08-22.md

```diff
@@ -0,0 +1,214 @@
+ 
+ 
+ 
+## Issue [#293](https://github.com/unicode-org/message-format-wg/issues/293 ) Spec clarification: selection without a function
+ 
+EAO: Summary: the issue proposes to disallow having selectors declared in a selection message that doesn’t specify its formatting function.
+ 
+MIH: Actually, ZBI is agreement with the proposal in the issue.
+ 
+EAO: One aspect that wasn’t clear to me is whether we’re talking about a limitation in the resulting data structure of the message or a limitation imposed on the syntax.
+ 
+MIH: I would say that we shouldn’t allow it. When I parse the message, I should expect to get an error there. If we enforce this in the syntax / data structures, it would make parsing messier. I would be fine just throwing an error in the parser.  One reason is because of ZBI’s point – we can always add it later, which would be a backwards compatible change.
+ 
+STA: My initial reaction is in line with ZBIs, but I am still thinking about it.
+ 
+EAO: Making this change would make it difficult if not impossible to convert Fluent into MessageFormat v2. Fluent has a structure in which a selector is implicitly typed, and it would make existing Fluent impossible to convert to MFv2 syntax in an automated manner.
+ 
+MIH: I would argue that the same problems of going from Fluent to MFv2 … because of that migration, we would be willing to pay the price of living with errors in lint. 
+ 
+ECH:
+
+1. Being consistent means requiring the formatting function for a selector, as we know we have to do for plurals
+2. How are Fluent messages not automatically upgradable to MFv2?
+ 
+STA: What are the reasons for forbidding this bare style of selectors?  How is that different from linting inside placeholders?
+ 
+MIH: In the issue itself, I explain the disadvantages. One is that if I see `match {$foo} * {...}`, I don’t know as a translator what type of case match values that correspond to the selector.  Second is I don’t know how to lint such messages. Third, is in Fluent, if I have a plural (number) match, and the input numerical value is 1, I don’t know to distinguish between `“1”` and `”ONE”`.  It creates a risky runtime feature.
+ 
+STA: If we have a risky runtime, then the runtime is broken.  For the problem in placeholders, we expect that everything should be stringifiable.  For the first issue, I can see the problem there.  I thought we were working on the assumption that most tools don’t let translators touch messages.
+ 
+EAO: ECH, in Fluent, you can have different variants in a selection message. The syntax of Fluent doesn’t require an annotation for what type of value selection should be performed on. Only at runtime is that type is known.  At runtime, we wrap values as a NumberValue for plural selects.  The actual behavior ends up depending on the runtime value. In Fluent, it’s not possible to expect the input value be of a certain type, so we don’t know if the input is the number 1 or if it’s the string `”ONE”`.
```

Comment:
```suggestion
EAO: ECH, in Fluent, you can have different variants in a selection message. The syntax of Fluent doesn’t require an annotation for what type of value selection should be performed on. Only at runtime is that type is known.  At runtime, we wrap values as a NumberValue for plural selects.  The actual behavior ends up depending on the runtime value. In Fluent, it’s not required to expect the input value be of a certain type, so we don’t know if the input is the number 1 or if it’s the string `”ONE”`.
```

We didn't cover this during the call, but Fluent does allow for using e.g. an explicit `NUMBER()` wrapper around a selector, which makes it always be handled as a number. So it's possible to have a Fluent message clearly communicate its expected input types, but it's not required.